dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -109,9 +107,23 @@ def new_general_foorm(survey_names:, day:, subject: FOORM_GENERAL_SURVEY_SUBJECT
     # Post workshop survey. This one will be emailed and displayed in the my PL page,
     # and can persist for more than a day, so it uses an enrollment code to be tied to a specific workshop.
     # GET /pd/workshop_survey/post/:enrollment_code
-    # This will direct to new_post_foorm
+    # If Build Your Own, redirect to its specific survey link. Otherwise, use new_post_foorm.
     def new_post
-      return new_post_foorm
+      course = get_workshop_by_enrollment_or_course_and_subject(enrollment_code: params[:enrollment_code], course: nil, subject: nil)&.course
+      if course == COURSE_BUILD_YOUR_OWN
+        redirect_to CDO.studio_url SURVEY_LINKS[:COURSE_BUILD_YOUR_OWN_TEACHER], CDO.default_scheme
+      else
+        return new_post_foorm
+      end
```

Comment:
what's the difference between redirect and return here? ie why does one path use one and the other uses something else? Just want to wrap my ahead around it :)

Reply:
`new_post_foorm` ends up returning a `render` for the `foorm` so I figured returning a `url` here wasn't going to work and I'd have to either do a render or redirect. I did so much tinkering I can't remember if I tested just a `return` for the url so let me try that super quick!

Reply:
Okay yep just double checked it's expecting a template so just a url doesn't work, it needs a `render` or `redirect`